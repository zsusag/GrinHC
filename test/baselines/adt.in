data Lst =
 | Nil
 | Cons Int Lst

let f :: Int -> Lst -> Lst = lambda (x::Int) :: Lst -> Lst =>
  lambda (ls::Lst) :: Lst => Cons x ls
in f 4 (Cons 3 (Cons 2 (Cons 1 (Nil))))
