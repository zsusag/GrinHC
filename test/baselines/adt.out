[("Nil",Lst),("Cons",(TArr (TArr Int -> Lst) -> Lst))]
Cons 4 Cons 3 Cons 2 Cons 1 Nil     
